packages/blade/src/components/Tag/_decisions/decisions.md

```diff
@@ -0,0 +1,208 @@
+# Tag 🏷️
+
+These are set of interactive keywords that help organise & categorise objects. Tags can be added or removed from an object by the users.
+
+<img width="426" alt="image" src="https://github.com/razorpay/blade/assets/30949385/75d8068d-d6fc-47e7-aa75-21686ed1c27d">
+
+## Design
+
+- [Tags - Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=29210-567576&t=HLRjz3OTofBtFdDl-0)
+- [Tags - Motion](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=29744-568562&mode=design&t=KI2KRJMzUpz8Ymr8-0)
+
+## Proposed API
+
+```jsx
+<Tag
+  icon={CheckIcon}
+  size="medium"
+  onDismiss={({ value }) => {
+    console.log('Close Icon Clicked', value);
+  }}
+>
+  Unpaid
+</Tag>
+```
+
+### Props
+
+```ts
+type TagProps = {
+  /**
+   * Decides the size of Tag
+   *
+   * @default medium
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Any icon from Blade
+   */
+  icon?: IconComponent;
+
+  /**
+   * Callback when close icon on Tag is clicked
+   *
+   * Gives a `value` argument which has same text as `children`
+   */
+  onDismiss?: ({ value }: { value: ReactText }) => void;
+
+  /**
+   * Text that renders inside Tag
+   */
+  children: ReactText;
+
+  /**
+   * Disable tag
+   */
+  isDisabled?: boolean;
+};
+```
+
+_Didn't consider any alternate API. Kept `size` and `children` similar to `<Badge />` and `onDismiss` similar to `<Alert />` and `<Dropdown />`_
+
+## Integration with Inputs
+
+<img width="358" alt="image" src="https://github.com/razorpay/blade/assets/30949385/71cd236a-93e7-46f0-9da6-12d519b958d4">
+
+Tags can be added inside other input components such as -
+
+- TextArea
+- SelectInput (in Dropdown)
+- TextInput
```

Comment:
Why do we need Tags in `<TextInput/>`, I'm wondering what is the use case for using tags inside text input

Reply:
So I have added image in Decision doc which has usecase of adding tags inside TextArea (Adding email id tags). Same usecase would also apply for TextInput where we don't want to show things on multiple lines